web/src/pm_list.ts

```diff
@@ -167,6 +168,10 @@ export function handle_narrow_activated(filter: Filter): void {
         }
         update_private_messages();
     }
+
+    if (!is_private_messages_in_view) {
```

Comment:
This should probably be `else if`, since otherwise we call this twice in the `narrow_to_private_messages_section` case.

Reply:
It isn't really the case. `narrow_to_private_messages_section` will work when narrowing into dm and the other one will only work when narrowing out from dm. Though it is indeed better to call it as `else if` since it may become problematic in future.